### PR TITLE
Fix deadlock on Kafka rebalance (when some worker threads are blocked)

### PR DESCRIPTION
**Pull request by tulinski**

### CODE REVIEW

File: src/main/java/com/rtbhouse/kafka/workers/impl/KafkaWorkersImpl.java [hunk 1]
```diff
@@ -104,12 +105,12 @@ public void start() {
         // - plus one consumer thread
         // - plus one punctuator thread
         final int allThreadsNum = workerThreadsNum + 2;
-        executorService = Executors.newFixedThreadPool(allThreadsNum);
-        executorService.execute(consumerThread);
+        executor = (ThreadPoolExecutor) Executors.newFixedThreadPool(allThreadsNum);
```
Comment (bdamian) [inline src/main/java/com/rtbhouse/kafka/workers/impl/KafkaWorkersImpl.java:108]: So maybe use `ThreadPoolExecutor::new`?

File: src/main/java/com/rtbhouse/kafka/workers/impl/consumer/ConsumerThread.java [hunk 2]
```diff
@@ -62,7 +62,7 @@ public ConsumerThread(
         super("consumer-thread", config, metrics, workers);
 
         this.consumerPollTimeout = Duration.ofMillis(config.getLong(WorkersConfig.CONSUMER_POLL_TIMEOUT_MS));
-        this.consumerProcessingTimeout = Duration.ofMillis(config.getLong(WorkersConfig.CONSUMER_PROCESSING_TIMEOUT_MS));
+        this.consumerProcessingTimeout = config.getConsumerProcessingTimeout();
```
Comment (bdamian) [inline src/main/java/com/rtbhouse/kafka/workers/impl/consumer/ConsumerThread.java:65]: It's nice! However, not consistent with calls next to it.

File: src/main/java/com/rtbhouse/kafka/workers/impl/task/TaskManager.java [hunk 3]
```diff
@@ -103,7 +111,8 @@ private void stopThreads() throws InterruptedException {
         // waits for all threads to stop because only then tasks can be rebalanced safely
         synchronized (rebalanceLock) {
             while (!allThreadsNotRunning()) {
-                rebalanceLock.wait();
+                rebalanceLock.wait(1_000L);
```
Comment (bdamian) [inline src/main/java/com/rtbhouse/kafka/workers/impl/task/TaskManager.java:114]: Would 1 second be enough?

File: src/main/java/com/rtbhouse/kafka/workers/impl/KafkaWorkersImpl.java (hunk 1 above)
Comment (tulinski) [reply src/main/java/com/rtbhouse/kafka/workers/impl/KafkaWorkersImpl.java:108]: Tak jest krócej i testy to pokrywają (a szansa, że się zwali jest minimalna).

File: src/main/java/com/rtbhouse/kafka/workers/impl/consumer/ConsumerThread.java (hunk 2 above)
Comment (tulinski) [reply src/main/java/com/rtbhouse/kafka/workers/impl/consumer/ConsumerThread.java:65]: Added getConsumerPollTimeout() method returning Duration.

File: src/main/java/com/rtbhouse/kafka/workers/impl/task/TaskManager.java (hunk 3 above)
Comment (tulinski) [reply src/main/java/com/rtbhouse/kafka/workers/impl/task/TaskManager.java:114]: We don't need to be super accurate in throwing timeout errors.

File: src/main/java/com/rtbhouse/kafka/workers/impl/consumer/ConsumerThread.java (hunk 2 above)
Comment (bdamian) [reply src/main/java/com/rtbhouse/kafka/workers/impl/consumer/ConsumerThread.java:65]: There is yet another one call below :)

Comment (tulinski) [reply src/main/java/com/rtbhouse/kafka/workers/impl/consumer/ConsumerThread.java:65]: Done
I thought it relates only to methods returning Duration

**Pull request merged**